tempesta_fw/sock.c

```diff
@@ -542,13 +542,6 @@ ss_tcp_process_data(struct sock *sk)
 			tp->copied_seq += count;
 			processed += count;
 
-			if (frag_list) {
-				off = 0;
-				skb = frag_list;
-				frag_list = frag_list->next;
-				skb->next = skb->prev = NULL;
-				goto repeat;
-			}
 			if (tcp_fin) {
```

Comment:
So if FIN is actually set in the last fragment we don't process all the fragments but first.